### DISCUSSION TITLE
Compare Two Image Sequences

### DISCUSSION THREAD

**Discussion by Jay123210599** [ImageMagick/ImageMagick#7791] [Help]
### ImageMagick version

7.1.1-40

### Operating system, version and so on

Windows 11

### Description

How do I use `magick compare` each frame from one image sequence to each frame from another, all at once? Both image sequences have at least 100 frames, the frames from the first sequence are all named primary, and the ones from the second sequence are all named secondary.

**Comment by fmw42** [comment]
You have to do them separately. There is no function to do them all at once.  So you would need to write a script FOR loop over each individual pair of frames.

**Comment by snibgo** [comment]
As fmw42 says.

However, if the frames all fit into memory at the same time (or you don't care about performance), you can easily establish whether all the corresponding pairs of input files are equal. First, make one frame for each pair of corresponding input frames. That frame is the difference between the input frames, and is entirely black if and only if the input frames are equal.

Then you "-compose lighten" to get a single image that has each channel in each pixel set to the highest value in all the difference frames. So that single frame is black if and only if all the pairs of input frames are equal. So the maximum value in that image is zero if and only if all the pairs of input frames are equal.

Windows BAT syntax:
```
magick ^
  apng:x0.apng NULL: apng:x1.apng ^
  -compose Difference -layers composite ^
  -compose Lighten ^
  -background Black -layers Flatten ^
  -channel RGB -format %[fx:maxima] ^
  info:
```
The output text is "0" if and only if if all the pairs of input frames are equal.
